### PR TITLE
Remove some unused code

### PR DESCRIPTION
**Pull request by wfleming**
Specifically, we no longer need the older `Analyzer::Config` or `Analyzer::EngineRegistry`. There were also a couple dangling `autoload` declarations for classes that had already been deleted.

### CODE REVIEW

File: spec/cc/analyzer/issue_spec.rb [hunk 1]
```diff
@@ -30,7 +17,7 @@ module CC::Analyzer
       it "adds a fingerprint when it is missing" do
         issue = Issue.new("", output.to_json)
 
-        expect(issue.fingerprint).to eq "433fae1189b03bcd9153dc8dce209fa5"
```
Comment (wfleming) [inline spec/cc/analyzer/issue_spec.rb]: These fingerprints were all based on reading the source of `lib/cc/analyzer/config.rb`, which I've deleted, so they changed here because I moved the relevant source to a new fixture file.

I am so mad that anybody ever thought it was a good idea to write tests that relied on an unrelated source file not changing.

File: spec/cc/analyzer/issue_spec.rb [hunk 2]
```diff
@@ -3,20 +3,7 @@
 module CC::Analyzer
   describe Issue do
     let(:output) do
-      {
```
Comment (wfleming) [inline spec/cc/analyzer/issue_spec.rb]: This data was already basically identical to the contents of `Factory#sample_issue`, may as well use it instead.

File: spec/support/factory.rb [hunk 3]
```diff
@@ -42,16 +42,16 @@ def create_yaml_with_no_engines
   def sample_issue(options = {})
     {
       "type" => "issue",
-      "check" => "Rubocop/Style/Documentation",
```
Comment (wfleming) [inline spec/support/factory.rb]: This factory has had the wrong key for a long, long time. Nice.

File: spec/cc/analyzer/issue_spec.rb (hunk 1 above)
Comment (ale7714) [reply spec/cc/analyzer/issue_spec.rb]: ohh :(
